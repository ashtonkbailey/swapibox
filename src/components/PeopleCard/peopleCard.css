article {
  color:  #FBF230;
  font-size: 16px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 6px;
  transition: background-color, opacity .6s ease-in-out;
}

article>img {
  margin: 1em;
}

.saber {
  height: 60px;
  width: 18%;
  position: relative;
  bottom: 16px;
  margin-top: 10px;
}

.saber:hover {
  /* opacity: 30; */
  cursor: pointer;
  transition: .3s ease-in-out;
  transform: scale(1.1);
}

.name {
  margin: 0;
  font-family: Roboto;
  font-style: normal;
  font-size: 40px;
  text-align: center;
  color: black;
  text-shadow:
  -1px -1px 0 rgb(238, 213, 19),  
  1px -1px 0 rgb(238, 213, 19),
  -1px 1px 0 rgb(238, 213, 19),
  1px 1px 0 rgb(238, 213, 19);
}

.small-card .name {
  font-size:  20px; 
}

.small-card {
  height: 500px;
  transition: background-color, opacity .6s ease-in-out;
  background: rgba(227, 218, 135, 0.2);
  /* max-width: 300px; */
  /* width: 80%; */
  /* padding-top: 20px; */
  margin-top: 60px;
  pointer-events: none;
}

.small-card>.saber {
  width: 60px;
}

.left-card {
  /* grid-column: 1/2; */
  opacity: .5;
  grid-row: 1;
  position: fixed;
  /* bottom: 210px; */
  width: 22%;
  left: 5%;
}

.right-card {
  grid-column: 3/4;
  opacity: .5;
  /* grid-row: 1; */
  position: fixed;
  width: 22%;
  /* bottom: 210px; */
  right: 5%;
}


.middle-card {
  /* grid-column: 2/3; */
  position: fixed;
  left: 32%;
  font-size: 24px;
  /* font-family: Verdana, Geneva, Tahoma, sans-serif; */
  /* min-width: 300px; */
  width: 34%;
  height: 500px;
  background: rgba(227, 218, 135, 0.4);
  border: 4px solid #EDE748;
  /* box-sizing: border-box; */
  box-shadow: 0px 4px 40px rgb(227, 218, 135);
  border-radius: 40px;
  margin: 20px;
  /* padding: 5px; */
  /* grid-row: 1; */
}

.small-card>img {
  width: 32%;
}

.middle-card>.saber {
  width: 16%;
  height: 16%;
}

.luke {
  border: solid 1px black;
  border-radius: 20px;
  box-shadow: 2px 2px  black;
  margin: 2px;
}

.left-card, .right-card {
  height: 400px;
  background: rgba(0, 0, 0, 0.66);
  border: 2px solid #EDE748;
  border-radius: 40px;
}

p {
  margin: 10px;
  color: black;
  text-shadow: 
  .5px .5px 0 rgb(238, 213, 19);
}
