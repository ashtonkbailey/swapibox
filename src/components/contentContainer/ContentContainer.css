// Film Crawl
.crawl{
    position:absolute;
    top:-3500px;
    bottom:0;
    left:0;
    right:0;
    overflow:hidden;
    text-align:center;
    font-family: 'Nanum Gothic', sans-serif;
}

.crawl>div{
    font-size:60px;
    padding-top:1000px;
    animation:autoscroll 1000s linear;
}

.crawl:after{
    content:'';
    position:absolute;
    top:0;
    bottom:0;
    left:0;
    right:0;
    background:linear-gradient(top,rgba(0,0,0,1),rgba(0,0,0,0) 100%);
    background:linear-gradient(to bottom,rgba(0,0,0,1),rgba(0,0,0,0) 100%);
    pointer-events:none;
}

.content-container{
    position:absolute;
    top:-160px;
    bottom:0;
    left:0;
    right:0;
    overflow:hidden;
    transform-origin:50% 100%;
    transform:perspective(400px) rotateX(20deg);
    color: yellow;
}

.title{
    position:absolute;
    bottom:0;
    left:0;
    right:0;
}

// People, Planets, Vehicles Cards
.card2 {
    width: 303px;
    height: 338px;
    background: rgba(0, 0, 0, 0.66);
border: 4px solid #EDE748;
box-sizing: border-box;
box-shadow: 0px 4px 40px #E3DA87;
border-radius: 66px;
}

@keyframes autoscroll{
    to {
        margin-top:-50000px;
    }    
}

.card1, .card3 {
    width: 24%;
    height: 240px;
    background: rgba(0, 0, 0, 0.66);
    border: 2px solid #EDE748;
    box-sizing: border-box;
    /* box-shadow: 0px 4px 40px #E3DA87; */
    border-radius: 66px;
}

.name {
font-family: Roboto;
font-style: normal;
font-size: 32px;
text-align: center;
color:  #FBF230;
}

.people-container {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    margin-top: 100px;
}

.component-container {
    display: flex;
    justify-content: space-between;
    flex-direction: row;
    align-items: center;
}

.arrow-button {
    height: 40px;
    width: 40px;
    margin: 50px 20px 0 20px;
    border-radius: 20px;
    font-size: 20px;
    font-family: 'Poller One', cursive;
    background-color: rgb(244, 217, 68);
    box-shadow: 10px 5px 5px rgba(255, 255, 255, 0.25), inset 15px 5px 3px rgba(255, 255, 255, 0.25);
}

.arrow-button:hover {
    background-color:  rgb(220, 210, 158);
    cursor: pointer;
}

.none {
    display: none;
}

