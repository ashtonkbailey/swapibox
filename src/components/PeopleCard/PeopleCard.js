import React from 'react';
import PropTypes from 'prop-types';

import lightsabers from '../../images/lightsabers.png';
import './peopleCard.css';
import luke from '../../images/luke.jpg';

const PeopleCard = ({
  card,
  carouselIndex,
  addToFavorites,
  chosenContent
}) => {
  let hiddenClass;
  const leftCardIndex = carouselIndex;
  let middleCardIndex = carouselIndex + 1;
  let rightCardIndex = carouselIndex + 2;

  if (carouselIndex === 9) {
    middleCardIndex = 0;
    rightCardIndex = carouselIndex - 8;
  }

  if (carouselIndex === 8) {
    rightCardIndex = carouselIndex - 8;
  }

  switch (card.index) {
    case middleCardIndex:
      hiddenClass = 'middle-card';
      break;
    case leftCardIndex:
      hiddenClass = 'left-card small-card';
      break;
    case rightCardIndex:
      hiddenClass = 'right-card small-card';
      break;
    default:
      hiddenClass = 'hidden';
  }
  
  if (chosenContent === 'people') {
    return (
      <article className={hiddenClass}>
        <h3 className="name">
          {card.name}
        </h3>
        <img
          src={luke}
          alt="character"
        />
        <p>
          species:
          {card.species}
        </p>
        <p>
          homeworld:
          {card.homeworld}
        </p>
        <p>
          population:
          {card.population}
        </p>
        <img
          className={ card.favorite ? "saber active" : "saber" }
          src={lightsabers}
          onClick={() => addToFavorites({ card })}
          alt="click to add to favorites"
        />
      </article>
    );
  }
  else if (chosenContent === 'planets') {
    return (
      <article className={hiddenClass}>
        <h3 className="name">
          {card.name}
        </h3>
        <img
          src={luke}
          className="luke"
          alt="planet"
        />
        <p>
          terrain:
          {card.terrain}
        </p>
        <p>
          climate:
          {card.climate}
        </p>
        <p>
          population:
          {card.population}
        </p>
        <p>
          residents:
          {card.residents}
        </p>
        <img
          className={ card.favorite ? "saber active" : "saber" }
          src={lightsabers}
          onClick={() => addToFavorites({ card })}
          alt="click to add to favorites"
        />
      </article>
    );
  }
    else if (chosenContent === 'vehicles') {
      return (
        <article className={hiddenClass}>
          <h3 className="name">
            {card.name}
          </h3>
          <img
            src={luke}
            alt="vehicle"
          />
          <p>
            model:
            {card.model}
          </p>
          <p>
            class:
            {card.class}
          </p>
          <p>
            passengers:
            {card.passengers}
          </p>
          <img
            className={ card.favorite ? "saber active" : "saber" }
            src={lightsabers}
            onClick={() => addToFavorites({ card })}
            alt="click to add to favorites"
          />
        </article>
      );
  }
};

PeopleCard.propTypes = {
  card: PropTypes.object.isRequired,
  carouselIndex: PropTypes.func,
  addToFavorites: PropTypes.func,
  chosenContent: PropTypes.array.isRequired,
};

export default PeopleCard;
