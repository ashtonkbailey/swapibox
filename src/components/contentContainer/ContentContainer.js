import React from 'react';
import PropTypes from 'prop-types';
import './ContentContainer.css';
import PeopleCard from '../PeopleCard/PeopleCard.js';

const ContentContainer = ({carouselIndex, incrementCarousel, decrementCarousel, film, contents}) => {
    let displayedContents;
    let containerClassName;
    let card2;
    let card3;

    if (carouselIndex < 9) {
        card2 = carouselIndex + 1;
    } else {
        card2 = carouselIndex - 9;
    }
    if (carouselIndex < 8) {
        card3 = carouselIndex + 2;
    } else {
        card3 = carouselIndex - 8;
    }

    if (!contents.length) {
        displayedContents = (
            <div>    
                <div className="crawl">
                    <div>
                    {film.crawl}
                    </div>
                </div>
                <h3 className="title"> {film.title} {film.year} </h3>
            </div>
        );
        containerClassName = "content-container";
    }
    else {
        const peopleCards = [
        <PeopleCard cardClass="small-card card1" card={contents[carouselIndex]} />, 
        <PeopleCard cardClass="card2" card={contents[card2]} />, 
        <PeopleCard cardClass="small-card card3" card={contents[card3]} /> ]
        displayedContents = (
            <div className="people-container">
                {peopleCards}
            </div>
        );
        containerClassName = "component-container";
    }

    return (
        <div className={containerClassName}>
            <button className={contents.length ? "arrow-button" : "none"} onClick={decrementCarousel}> <i className="fas fa-arrow-left"></i> </button>
            {displayedContents}
            <button className={contents.length ? "arrow-button" : "none"} onClick={incrementCarousel}> <i className="fas fa-arrow-right"></i> </button>
         </div>
    )
}

export default ContentContainer;
