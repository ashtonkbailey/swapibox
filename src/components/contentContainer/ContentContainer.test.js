import React from 'react'; 
import { shallow, mount } from 'enzyme';
import ContentContainer from './ContentContainer.js';
import PeopleCard from '../PeopleCard/PeopleCard.js'

describe ('content container', () => {
 

  let mockCard;
  let mockFilm;
  let mockIndex;
  let mockDisplayedContent;
  let wrapper;
  let mockInitialContent;

  beforeEach(() => {
     mockCard = {
      favorite: false,
      homeworld: "Tatooine",
      index: 0,
      name: "Luke Skywalker",
      population: "200000",
      species: "Human"
    }
     mockFilm = {
      crawl: "It is a dark time for the Rebellion. Although the Death Star has been destroyed, Imperial troops have driven the Rebel forces from their hidden base and pursued them across the galaxy.  Evading the drâ€¦",
      title: "The Empire Strikes Back",
      year: "1980"  
    }
     mockIndex = 0;
     mockDisplayedContent = [
      mockCard, 
      {
        favorite: false,
        homeworld: "Tatooine",
        index: 1,
        name: "C-3PO",
        population: "200000",
        species: "Droid"
      },
      {
        favorite: false,
        homeworld: "Naboo",
        index: 2,
        name: "R2-D2",
        population: "4500000000",
        species: "Droid"
      }
    ];
    mockInitialContent = [];
    wrapper = mount (<ContentContainer film={mockFilm}
      contents={mockDisplayedContent}
      addToFavorites = {jest.fn()}
      incrementCarousel = {jest.fn()}
      carouselIndex = {mockIndex}
      key={0}
    /> );
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renders all of the contents', () => {
    expect(wrapper.find(PeopleCard).length).toEqual(3);
  })
  it ('should render the film scroll if no contents are passed in', () => {
    wrapper = mount (<ContentContainer film={mockFilm}
      contents={mockDisplayedContent}
      addToFavorites = {jest.fn()}
      incrementCarousel = {jest.fn()}
      carouselIndex = {mockIndex}
      key={0}
    /> )
  })
})