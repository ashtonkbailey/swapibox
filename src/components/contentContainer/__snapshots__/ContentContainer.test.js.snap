// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content container should match the snapshot 1`] = `undefined`;
